imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -236,10 +238,22 @@ def get_variable_attributes(self, variable_name: str) -> dict:
         Returns
         -------
         dict
-            I have no idea todo check.
+            Information containing specific variable attributes
+            associated with "variable_name".
         """
-        # TODO: Create a variable attribute schema file, validate here
-        if variable_name in self.variable_attributes:
-            return self.variable_attributes[variable_name]
-        # TODO: throw an error?
-        return {}
+        output = dict()
+        for attr_name in self.variable_attribute_schema.keys():
+            if variable_name in self._variable_attributes:
+                output = self._variable_attributes[variable_name]
+            elif (
+                variable_name is not None
+                and attr_name in self._variable_attributes["variable_name"]
+            ):
+                output = self._variable_attributes["variable_name"][attr_name]
+            elif (
+                variable_name not in self.variable_attribute_schema
```

Comment:
Rather than checking `variable_name` against the schema, we should check the attributes against the schema. So you would rewrite this block as something like:

```
for attribute in self.variable_attribute_schema.keys():
  if attribute in self._variable_attributes[variable_name]:
    add to output
  if attribute not in self._variable_attributes[variable_name] and variable is required:
    add to output as None
```

Reply:
Hahah, yes, thank you so much! I caught this momentarily after I submitted the PR request. Felt very silly. I should have probably started this one as a draft.